script/hipify-inplace.sh

```diff
@@ -51,23 +61,36 @@ echo "Logs written to ${HIPIFY_LOG}"
 
 # Additional fixes for project-specific things and things hipify misses or gets
 # wrong.
-for src in ${srcs[@]}; do
-    sed -i 's@#include <hipblas.h>@#include <hipblas/hipblas.h>@' $src
-    sed -i 's@#include <hipsparse.h>@#include <hipsparse/hipsparse.h>@' $src
-    sed -i 's@#include <cuda_fp8.h>@#include <hip/hip_fp8.h>@' $src
-    sed -i 's@#include <cuda_bf16.h>@#include <hip/hip_bf16.h>@' $src
-    sed -i 's@\bDGL_USE_CUDA\b@DGL_USE_ROCM@g' $src
-    sed -i 's@\bCUB_VERSION\b@HIPCUB_VERSION@g' $src
-    sed -i 's@\bCUDART_ZERO_BF16\b@HIPRT_ZERO_BF16@g' $src
-    sed -i 's@\bCUDART_INF_BF16\b@HIPRT_INF_BF16@g' $src
-    sed -i 's@\bthrust::cuda::par@thrust::hip::par@g' $src
-    sed -i 's@\b__nv_fp8_e4m3\b@__hip_fp8_e4m3@g' $src
-    sed -i 's@\b__nv_fp8_e5m2\b@__hip_fp8_e5m2@g' $src
-    sed -i 's@\bCUBLAS_GEMM_DEFAULT_TENSOR_OP\b@HIPBLAS_GEMM_DEFAULT@g' $src
-    sed -i 's@\bcurand4\b@hiprand4@g' $src
-    # hipify uses the old one
-    sed -i 's@\bhip_bfloat16\b@__hip_bfloat16@g' $src
-    sed -i 's@\b__trap();@abort();@' $src
+for src in ${all_srcs[@]}; do
+    sed -i -e 's@#include <hipblas.h>@#include <hipblas/hipblas.h>@' \
+        -e 's@#include <hipsparse.h>@#include <hipsparse/hipsparse.h>@' \
+        -e 's@#include <cuda_fp8.h>@#include <hip/hip_fp8.h>@' \
+        -e 's@#include <cuda_bf16.h>@#include <hip/hip_bf16.h>@' \
+        -e 's@#include <cub/cub.cuh>@#include <hipcub/hipcub.hpp>@' \
+        -e 's@#include "hip/extension/@#include "./cuda/extension/@' \
+        -e 's@#include "hip/cooperative_minibatching_utils.h"@#include "./cuda/cooperative_minibatching_utils.h"@' \
+        -e 's@#include "hip/max_uva_threads.h"@#include "./cuda/max_uva_threads.h"@' \
+        -e 's@\.\./hip/@@' \
+        -e 's@\bDeviceFor::Bulk\b@Bulk@g' \
+        -e 's@\bcub::@hipcub::@g' \
+        -e 's@\bcuda::stream_ref@cuco::cuda_stream_ref@' \
+        -e 's@\bcuda::::min@min@' \
+        -e 's@\bcuda::proclaim_return_type@proclaim_return_type@' \
+        -e 's@\bDGL_USE_CUDA\b@DGL_USE_ROCM@g' \
+        -e 's@\b__CUDA_ARCH__\b@__HIP_DEVICE_COMPILE__@g' \
+        -e 's@\bGRAPHBOLT_USE_CUDA\b@GRAPHBOLT_USE_ROCM@g' \
+        -e 's@\bCUB_VERSION\b@HIPCUB_VERSION@g' \
+        -e 's@\bCUDART_ZERO_BF16\b@HIPRT_ZERO_BF16@g' \
+        -e 's@\bCUDART_INF_BF16\b@HIPRT_INF_BF16@g' \
+        -e 's@\bthrust::cuda::par@thrust::hip::par@g' \
+        -e 's@\b__nv_fp8_e4m3\b@__hip_fp8_e4m3@g' \
+        -e 's@\b__nv_fp8_e5m2\b@__hip_fp8_e5m2@g' \
+        -e 's@\bCUBLAS_GEMM_DEFAULT_TENSOR_OP\b@HIPBLAS_GEMM_DEFAULT@g' \
+        -e 's@\bcurand4\b@hiprand4@g' \
+        -e 's@\bhip_bfloat16\b@__hip_bfloat16@g' `# hipify uses the old one` \
```

Comment:
At this point, it might be easier to write a heredoc-style inline sed script file. This would avoid all the newline escaping and execution of comments and such.

Reply:
Done.